modules/server/src/main/scala/io/chrisdavenport/ember/server/EmberServerBuilder.scala

```diff
@@ -0,0 +1,128 @@
+package io.chrisdavenport.ember.server
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import fs2.concurrent._
+import org.http4s._
+import org.http4s.server.Server
+import scala.concurrent.duration._
+import java.net.InetSocketAddress
+import java.util.concurrent.Executors
+import java.nio.channels.AsynchronousChannelGroup
+
+
+final class EmberServerBuilder[F[_]: Concurrent: Timer: ContextShift] private (
+  private val host: String,
+  private val port: Int,
+  private val httpApp: HttpApp[F],
+  private val agR: Resource[F, AsynchronousChannelGroup],
+  private val onError: Throwable => Response[F],
+  private val onWriteFailure : Option[(Option[Request[F]], Response[F], Throwable) => F[Unit]],
```

Comment:
This type makes my head hurt.  The outer option in particular.  Could it not default to `Function.const(F.unit)`?

Reply:
Didn't like `Function.const`, but I could lose the Option.